### PR TITLE
Improve translation of milestone filters

### LINKED ISSUE #26567: Milestone due sorting is confusing
### Description

The terms `closest` and `furthest` don't describe the actual sorting behavior as these two are semantically relative to the current date.

Could we switch to `earliest` and `latest` instead?

### Screenshots

With 17.08.2023 being the current date, `Tomorrow` would be considered the milestone with the "Closest due date".

image

### Gitea Version

1.20.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

_No response_

### Browser Version

Firefox 116

### PR DESCRIPTION
**Pull request by CaiCandong**
https://github.com/go-gitea/gitea/issues/26567#issue-1855312074
> The terms `closest` and `furthest` don't describe the actual sorting behavior as these two are semantically relative to the current date.
> Could we switch to `earliest` and `latest` instead?

close #26567

# Screenshot
-  Earliest due date
image
- Latest due date
image

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -1815,8 +1815,8 @@ milestones.edit_success = Milestone "%s" has been updated.
 milestones.deletion = Delete Milestone
 milestones.deletion_desc = Deleting a milestone removes it from all related issues. Continue?
 milestones.deletion_success = The milestone has been deleted.
-milestones.filter_sort.closest_due_date = Closest due date
-milestones.filter_sort.furthest_due_date = Furthest due date
```
Comment (wolfogre) [inline options/locale/locale_en-US.ini:1819]: Please be careful, I think otbutz‘s idea is "Closest due date -> Earliest due date", not " -> Latest". You ignored "due data" and got a wrong match.

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -1815,8 +1815,8 @@ milestones.edit_success = Milestone "%s" has been updated.
...
+milestones.filter_sort.closest_due_date = Latest
+milestones.filter_sort.furthest_due_date = Earliest
```
Comment (wolfogre) [inline options/locale/locale_en-US.ini]: Please be careful, I think otbutz‘s idea is `Closest due date -> Earliest due date`, not ` -> Latest`. You ignored "due data" and got a wrong match.

Comment (CaiCandong) [reply options/locale/locale_en-US.ini]: My apologies. Thanks for the tip.

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -1815,8 +1815,8 @@ milestones.edit_success = Milestone "%s" has been updated.
 milestones.deletion = Delete Milestone
 milestones.deletion_desc = Deleting a milestone removes it from all related issues. Continue?
 milestones.deletion_success = The milestone has been deleted.
-milestones.filter_sort.closest_due_date = Closest due date
-milestones.filter_sort.furthest_due_date = Furthest due date
+milestones.filter_sort.closest_due_date = Earliest due date
+milestones.filter_sort.furthest_due_date =  Latest due date
```
Comment (yp05327) [inline options/locale/locale_en-US.ini]:
```suggestion
milestones.filter_sort.furthest_due_date = Latest due date
```

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Should rename the key as well.

**Review by lunny — APPROVED** [review]

**Pull request merged**